### DISCUSSION TITLE
Add podLabel options to values.yaml

### DISCUSSION THREAD

**Discussion by kubblai** [istio/istio#49270] [Ideas]
Hi Guys

We use kubecost in our org and its really nice to be able to use this feature to add squad labels to pods to manage/monitor costs:

https://github.com/istio/istio/blob/7da81251e0bd67a7be5a2449884a0c13544a1dc6/manifests/charts/istio-control/istio-discovery/values.yaml#L84C5-L85C18

Unfortunately this option if not available for:

https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/values.yaml
and
https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-egress/values.yaml

Would be great if this could be added to the templates for these charts too so we can monitor costs like this https://github.com/istio/istio/blob/7da81251e0bd67a7be5a2449884a0c13544a1dc6/manifests/charts/istio-control/istio-discovery/templates/deployment.yaml#L47

**Comment by husnialhamdani** [comment]
What about this one https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/values.yaml#L7 ?, 
I think you can add from there

**Reply by kubblai** [reply]
that writes it all over the place including to the selector.matchlabels which then breaks helm upgrade as the field is immutable
